lib/asm-parser-vc.js

```diff
@@ -0,0 +1,256 @@
+        this.commentOnly = /^;/;
+        this.filenameComment = /^; File (.+)/;
+        this.lineNumberComment = /^; Line ([0-9]+)/;
+        this.beginSegment = /^(CONST|_BSS|[prx]?data(\$[a-zA-Z]+)?|_TEXT|text(\$[a-zA-Z]+)?)\s+SEGMENT/;
+        this.endSegment = /^(CONST|_BSS|[prx]?data(\$[a-zA-Z]+)?|_TEXT|text(\$[a-zA-Z]+)?)\s+ENDS/;
+        this.beginFunction = /(^; Function compile flags: )|(^_TEXT\s+SEGMENT)|(^[a-zA-Z@$?_][a-zA-Z0-9@$?_<>]*\s+PROC)/;
+        this.endFunction = /^([a-zA-Z@$?_][a-zA-Z0-9@$?_<>]*)\s+ENDP/;
+
+        this.labelDef = /^([a-zA-Z@$?_][a-zA-Z0-9@$?_<>]*)\s+(PROC|=|D[BWDQ])/;
+        this.definesGlobal = /^(PUBLIC|EXTRN)\s+/;
+        this.definesFunction = /^([a-zA-Z@$?_][a-zA-Z0-9@$?_<>]*)\s+PROC/;
+        this.labelFind = /[a-zA-Z@$?_][a-zA-Z0-9@$?_<>]*/g;
+        this.dataDefn = /^([a-zA-Z@$?_][a-zA-Z0-9@$?_<>]* D[BWDQ]\s|\s+D[BWDQ]\s|\s+ORG)/;
+
+        // these are set to an impossible regex, because VC doesn't have inline assembly
+        this.startAppBlock = this.startAsmNesting = /a^/;
+        this.endAppBLock = this.endAsmNesting = /a^/;
+        // same, but for CUDA
+        this.cudaBeginDef = /a^/;
+    }
+
+    hasOpcode(line) {
+        // note: cl doesn't output leading labels
+        // strip comments
+        line = line.split(';', 1)[0];
+        // check for empty lines
+        if (line.length === 0) return false;
+        // check for a local label definition
+        if (line.match(this.localLabelDef)) return false;
+        // check for global label definitions
+        if (line.match(this.definesGlobal)) return false;
+        // check for data definitions
+        if (line.match(this.dataDefn)) return false;
+        // check for segment begin and end
+        if (line.match(this.beginSegment) || line.match(this.endSegment)) return false;
+        // check for function begin and end
+        // note: functionBegin is used for the function compile flags comment
+        if (line.match(this.definesFunction) || line.match(this.endFunction)) return false;
+        // check for miscellaneous directives
+        if (line.match(this.miscDirective)) return false;
+        
+        return !!line.match(this.hasOpcodeRe);
+    }
+
+    labelFindFor() {
+        return this.labelFind;
+    }
+
+    processAsm(asm, filters) {
+        const getFilenameFromComment = line => {
+            const matches = line.match(this.filenameComment);
+            if (!matches) {
+                return null;
+            } else {
+                return matches[1];
+            }
+        };
+        const getLineNumberFromComment = line => {
+            const matches = line.match(this.lineNumberComment);
+            if (!matches) {
+                return null;
+            } else {
+                return parseInt(matches[1]);
+            }
+        };
+
+        const asmLines = utils.splitLines(asm);
+        // note: VC doesn't output unused labels, afaict
```

Comment:
If this is true, there are ways to lock a filter to ON/OFF on the UI, which might be useful here

Reply:
I don't know how to do this